File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Model/Statements/CompStmt.java
package Model.Statements;

import Exceptions.MyControllerException;
import Model.Utils.MyIStack;
import Model.PrgState;

public class CompStmt implements IStmt
{
    IStmt first;
    IStmt snd;
    public CompStmt(IStmt first,IStmt snd){
        this.first=first;
        this.snd=snd;
    }

    public String toString() {
        return first+";"+snd;
    }
    public PrgState execute(PrgState state)throws MyControllerException
    {
        MyIStack<IStmt> stk = state.getStk();
        stk.push(snd);
        stk.push(first);
        return null;
    }
}

File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/C#LabAssignment/sharplab/Model/Statements/CompStmt.cs
using sharplab.Model.Utils;
using System;
using System.Collections.Generic;
using System.Text;

namespace sharplab.Model.Statements
{
    class CompStmt : IStmt
    {
        IStmt first;
        IStmt snd;
        public CompStmt(IStmt first, IStmt snd)
        {
            this.first = first;
            this.snd = snd;
        }

        public override string ToString()
        {
            return first + ";" + snd;
        }

        public PrgState Execute(PrgState state)
        {
            IMyStack<IStmt> stk = state.GetStk();
            stk.Push(snd);
            stk.Push(first);
            return state;
        }
    }
}
